crates/rspack_core/src/resolver/factory.rs

```diff
@@ -4,14 +4,12 @@ use dashmap::DashMap;
 use rustc_hash::FxHasher;
 
 use super::resolver_impl::Resolver;
-use crate::DependencyType;
 use crate::{DependencyCategory, Resolve};
 
 #[derive(Debug, Hash, Eq, PartialEq, Clone)]
 pub struct ResolveOptionsWithDependencyType {
```

Comment:
```suggestion
pub struct ResolveOptionsWithDependencyCategory {
```

Reply:
It's align to webpack: https://github.com/webpack/webpack/blob/87660921808566ef3b8796f8df61bd79fc026108/lib/ResolverFactory.js#L24

Reply:
I know, but ResolveOptionsWithDependencyType don't have dependency_type field feels wired, so add some comments if you want keep the name